components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1467,11 +1474,63 @@ private void setIdToken(OAuth2AuthorizeRespDTO authzRespDTO,
-                tokenBindingValue);
+                tokenBindingValue, tokenExtendedAttributes);
+    }
+
+    private AccessTokenExtendedAttributes getExtendedTokenAttributes(OAuthMessage oAuthMessage,
+                                                                     OAuth2Parameters oauth2Params) {
+
+        try {
+            ServiceProvider serviceProvider = getServiceProvider(oauth2Params.getClientId());
+            // TODO: Improve to read the script separately instead of reading from adaptive script.
+            if (!EndpointUtil.isExternalizedConsentPageEnabledForSP(serviceProvider) ||
+                    serviceProvider.getLocalAndOutBoundAuthenticationConfig().getAuthenticationScriptConfig() == null) {
+                return null;
+            }
+            JSEngine jsEngine = EngineUtils.getEngineFromConfig();
+            JsLogger jsLogger = new JsLogger();
+            Map<String, Object> bindings = new HashMap<>();
+            bindings.put(FrameworkConstants.JSAttributes.JS_LOG, jsLogger);
+            List<String> accessTokenJSObject = new ArrayList<>();
+            ObjectMapper mapper = new ObjectMapper();
+            Map<String, Object> parameterMap =
+                    mapper.convertValue(oAuthMessage.getRequest().getParameterMap(), Map.class);
+            accessTokenJSObject.add(ACCESS_TOKEN_JS_OBJECT);
+            Map<String, Object> result = jsEngine
+                    .createEngine()
+                    .addBindings(bindings)
+                    .evalScript(
+                            serviceProvider.getLocalAndOutBoundAuthenticationConfig().getAuthenticationScriptConfig()
+                                    .getContent())
+                    .invokeFunction(DYNAMIC_TOKEN_DATA_FUNCTION, parameterMap)
+                    .getJSObjects(accessTokenJSObject);
+            ObjectMapper objectMapper = new ObjectMapper();
+            String json = objectMapper.writeValueAsString(result.get(ACCESS_TOKEN_JS_OBJECT));
+            AccessTokenExtendedAttributes
+                    accessTokenExtendedAttributes = objectMapper.readValue(json, AccessTokenExtendedAttributes.class);
+            if (accessTokenExtendedAttributes != null) {
+                accessTokenExtendedAttributes.setExtendedToken(true);
+            }
+            return accessTokenExtendedAttributes;
+        } catch (Exception e) {
+            String msg = "Error occurred when processing consent response request from tenant: " +
+                    oauth2Params.getTenantDomain() + " application: " + oauth2Params.getClientId() + "after consent.";
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                LoggerUtils.triggerDiagnosticLogEvent(OAuthConstants.LogConstants.OAUTH_INBOUND_SERVICE, null,
+                        OAuthConstants.LogConstants.FAILED, msg, "authorize-client", null);
+            }
+            log.warn(msg, e);
```

Comment:
This can be caused due to the invalid scripts from the organization. Therefore, instead of breaking the flow, decided to publish Diagnostic Logs and continue the flow.